pkg/asset/cluster/azure/azure.go

```diff
@@ -38,7 +39,42 @@ func PreTerraform(ctx context.Context, clusterID string, installConfig *installc
 	if err != nil {
 		return errors.Wrap(err, "failed to get session")
 	}
-	return tagResourceGroup(ctx, clusterID, installConfig, session)
+
+	if err := tagResourceGroup(ctx, clusterID, installConfig, session); err != nil {
+		return err
+	}
+
+	if len(installConfig.Config.Azure.VirtualNetwork) == 0 {
+		return nil
+	}
+	return tagVNet(ctx, clusterID, installConfig, session)
```

Comment:
nit: while this is elegant it might be easier to extend this function later if you do:

```go
if err := tagVnet(ctx, clusterID, installConfig, session); err != nil {
    return nil
}

return nil
```

Reply:
done

Reply:
Golint did not like it:
```
 pkg/asset/cluster/azure/azure.go:43:2: if-return: redundant if ...; err != nil check, just return error instead. (revive)
	if err := tagVNet(ctx, clusterID, installConfig, session); err != nil {
		return err
	} 
```

Reply:
found a middle ground that the linter was happy with:
```golang
err = tagVnet(ctx, clusterID, installConfig, session)
return err
```

Reply:
If you prefer to keep the `if ...` version, you can also disable the linter
```
if err := tagVNet(ctx, clusterID, installConfig, session); err != nil { //nolint:revive // we prefer it this way rather than return the functions result
```